garrysmod/lua/includes/modules/widget.lua

```diff
@@ -8,38 +8,38 @@ widgets = {}
 
 --
 -- Holds the currently hovered widget
--- 
+--
 widgets.Hovered = nil
 widgets.HoveredPos = Vector( 0, 0, 0 )
 
 --
 -- Holds the current pressed widget
--- 
+--
 widgets.Pressed = nil
 
 local function UpdateHovered( pl, mv )
 
 	if ( !IsValid( pl ) ) then return end
 
-	if ( !pl:Alive() ) then 
-		pl:SetHoveredWidget( NULL ) 
-		return 
+	if ( !pl:Alive() ) then
+		pl:SetHoveredWidget( NULL )
+		return
 	end
 
 	local OldHovered = pl:GetHoveredWidget()
 	pl:SetHoveredWidget( NULL )
 
-	local trace = 
+	local trace =
 	{
 		start	= pl:EyePos(),
 		endpos	= pl:EyePos() + pl:GetAimVector() * 256,

```

Comment:
someone should optimizz this to 
```Lua
endpos = pl:GetAimVector(),
...

endpos:Mult(256)
endpos:Add(ply:EyePos())
```

Reply:
Hmm, why not?

endpos = pl:GetAimVector(),
...

trace.endpos:Mult(256)
trace.endpos:Add(trace.start)

Reply:
and i little dont understand you.

Reply:
it was pseudo-ish code, obviously you can do what you showed

Reply:
It's faster to do what I dod because i directly edits the vectors without recreating them